microsoft-edge/webview2/how-to/self-hosting.md

```diff
@@ -0,0 +1,202 @@
+
+
+<!-- ------------------------------ -->
+#### Option 1: Manually install preview channels on machines
+
+Insider channels can be manually installed through the following links:
+
+| Channel | Link |
+| --- | --- |
+| Canary | [Download](https://go.microsoft.com/fwlink/?linkid=2084649&Channel=Canary&language=en) |
+| Dev | [Download](https://go.microsoft.com/fwlink/?linkid=2093291) |
+| Beta | [Download](https://go.microsoft.com/fwlink/?linkid=2093376) |
+
+This only needs to be done once per machine.  Prerelease channels are evergreen, so they will automatically get updated when newer versions are available.
+
+
+<!-- ------------------------------ -->
+#### Option 2: Programmatically deploy preview channels via API endpoints
+
+In your app's code, periodically poll the following API endpoints to get and deploy the latest version of each preview channel of Microsoft Edge:
+
+| Channel | Link |
+| --- | --- |
+| Canary | [https://edgeupdates.microsoft.com/api/products/canary](https://edgeupdates.microsoft.com/api/products/canary)<br>[MSI Link](https://go.microsoft.com/fwlink/?linkid=2084649&Channel=Canary&language=en)|
+| Dev | [https://edgeupdates.microsoft.com/api/products/dev](https://edgeupdates.microsoft.com/api/products/dev) |
+| Beta | [https://edgeupdates.microsoft.com/api/products/beta](https://edgeupdates.microsoft.com/api/products/beta) |
+
+The Edge Dev and Edge Beta channels contain MSI Links.  The Edge Canary channel has a separate MSI link.
+
+
+<!-- ====================================================================== -->
+## Diagnose potential issues with the WebView2 Runtime
+
+To diagnose or troubleshoot potential prerelease WebView2 Runtime issues that are revealed by your app, use the following techniques.
+
+
+<!-- ------------------------------ -->
+#### Confirm that your app is running a preview channel of WebView2
+
+1. Open Task Manager
+
+1. Find your application's process, then find the Edge WebView2 child process
+
+1. Right click on the Edge WebView2 process, and then `Open file location`. The folder will be version number of the runtime.
+
+<!-- ------------------------------ -->
+#### Switch to the Stable WebView2 Runtime channel to see if the issue reproduces
+
+If the issue doesn't reproduce on the Stable channel of the WebView2 Runtime, it's likely that the issue is with the preview channel of WebView2, or how your app interacts with the preview channel of WebView2.
+
+You can easily switch to the stable channel (Stable WebView2 Runtime) by using environmental variables, because revised environment variables override the registry settings (which are set by group policies).  This allows you to easily switch from a preview channel to the stable channel, to see if the issue reproduces.  If you are already using environmental variables, you simply update the values.  Note that changing the environment variables changes the WebView2 channel for all apps that use WebView2 on the machine.
```

Comment:
Updated the wording to reflect that:
`Note that if you are setting the environment variable globally, it will change the WebView2 channel for all apps that use WebView2 on the machine. `